### DISCUSSION TITLE
Shazam support for getting streaming track info

### DISCUSSION THREAD

**Discussion by DDSchrader** [ciderapp/Cider#1597] [Feature Request]
I would like to see Shazam support added, perhaps as a plugin, that could be enabled when streaming audio from sources that don't supply track info beyond the title and artist. The information and cover art should be used in place of the fixed icon of the streamed source.

**Comment by GMen0n** [comment]
isnt it possible to just get all these functions with the shazam app or the shazam browser extension , if you are ever streaming anything on cider you already get song info so i really dont see why cider would need it right , neat feature nevertheless

**Reply by DDSchrader** [reply]
Yes, these functions are available via the Shazam app or browser extension, but these rely on the machine's microphone to identify the song and the data found will only appear in the app or browser. Having this integrated into Cider would allow it to fill in missing metadata, cover art, lyrics, etc. 

Just to reinforce the point I made about using the microphone, this doesn't allow song detection at low playback levels or during headphone use.  I wrote some code for my own use with MPD/JACK that reads the incoming stream into Shazam after each track starts and then displays the cover art and track info in the playback app's screen. But that doesn't help me when using the much better Cider app.
